monai/data/utils.py

```diff
@@ -588,11 +588,11 @@ def decollate_batch(batch, detach: bool = True, pad=True, fill_value=None):
         if isinstance(batch, MetaObj):
             for t, m in zip(out_list, decollate_batch(batch.meta)):
                 if isinstance(t, MetaObj):
-                    t.meta = m
+                    t.meta = deepcopy(m)
```

Comment:
May I know why add `deepcopy` here? I think the decollated items of `batch.meta` should be dispatched to every item in `out_list`? Or Maybe I misundertand this logic here.

Thanks.

Reply:
this is to ensure all the samples in a batch don't share the meta info (to be consistent with applied_operations). I think it's robust, and doesn't require much cpu time as it's done only once per tensor.

Reply:
thanks -- that's removed in the latest commit